src/test/java/org/pesmypetcare/webservice/service/FreqTrainingServiceTest.java

```diff
@@ -0,0 +1,132 @@
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class FreqTrainingServiceTest {
+
+    private static List<Map<String, Object>> freqTrainingList;
+    private static FreqTrainingEntity freqTrainingEntity;
+    private static String owner;
+    private static String petName;
+    private static String date;
+    private static String date2;
+    private static Double value;
+
+    @Mock
+    private FreqTrainingDao freqTrainingDao;
+
+    @InjectMocks
+    private FreqTrainingService service = new FreqTrainingServiceImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        freqTrainingList = new ArrayList<>();
+        freqTrainingEntity = new FreqTrainingEntity();
+        date = "2020-02-13T10:30:00";
+        date2 = "2021-02-13T10:30:00";
+        owner = "TupoJohn";
+        petName = "Perico";
+        value = 9.0 / 2.0;
+    }
+
+    @Test
+    public void shouldReturnNothingWhenFreqTrainingCreated() {
+        service.createFreqTraining(owner, petName, date, freqTrainingEntity);
+        verify(freqTrainingDao).createFreqTraining(isA(String.class), isA(String.class), isA(String.class),
+            isA(FreqTrainingEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenFreqTrainingDeleted() throws DatabaseAccessException {
+        service.deleteFreqTrainingByDate(owner, petName, date);
+        verify(freqTrainingDao).deleteFreqTrainingByDate(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllFreqTrainingsDeleted() throws DatabaseAccessException {
+        service.deleteAllFreqTrainings(owner, petName);
+        verify(freqTrainingDao).deleteAllFreqTrainings(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllFreqTrainingDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(freqTrainingDao).deleteAllFreqTrainings(any(String.class),
+                isA(String.class));
+            service.deleteAllFreqTrainings(owner, petName);
+        }, "Should return a database access exception when a freqTraining deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMealEntityWhenFreqTrainingRetrieved() throws DatabaseAccessException {
+        when(freqTrainingDao.getFreqTrainingByDate(owner, petName, date)).thenReturn(freqTrainingEntity);
+        FreqTrainingEntity meal = service.getFreqTrainingByDate(owner, petName, date);
+        assertSame(freqTrainingEntity, meal, "Should return a freqTraining entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetFreqTrainingRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(freqTrainingDao).getFreqTrainingByDate(any(String.class), any(String.class),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 132).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)